web3/_utils/filters.py

```diff
@@ -186,6 +186,7 @@ class TransactionFilter(Filter):
 class LogFilter(Filter):
     data_filter_set = None
     data_filter_set_regex = None
+    data_filter_set_function = None
```

Comment:
This could be removed from this context if a reviewer feels strongly about it. I just came across a pycharm warning that this is set below but never initialized to begin with.